fish-rust/src/future_feature_flags.rs

```diff
@@ -134,36 +106,57 @@ const metadata: [FeatureMetadata; 4] = [
 ];
 
 /// The singleton shared feature set.
-static mut global_features: Features = Features::new();
+static FEATURES: RwLock<Features> = RwLock::new(Features::new());
```

Comment:
I don't see a reason for using locking instead of atomic bools.
(Also our current stance is to use the simpler Mutex unless we really need concurrent multiple readers)

IIRC feature flags are set early on during startup and never changed.
From that PoV, atomic bools make most sense because they are usually zero-overhead

Reply:
TBH this change can be `#[cfg(test)]`'d. I don't see the point arguing about the performance for functions that are called very rarely. This is a `RwLock` since we need to guarantee that a test that's dependant on one particular feature flag actually has that value during the test. `AtomicBool` does not work for that, and neither does a `Mutex`, unless I am misunderstanding how you are suggesting to use the mutex.